ionc/ion_timestamp.c

```diff
@@ -1074,25 +1084,34 @@ iERR _ion_timestamp_equals_helper(const ION_TIMESTAMP *ptime1, const ION_TIMESTA
         if (HAS_TZ_OFFSET(ptime1) ^ HAS_TZ_OFFSET(ptime2)) {
             goto is_false;
         }
+        if (HAS_TZ_OFFSET(ptime1) && (ptime1->tz_offset != ptime2->tz_offset)) {
+            goto is_false;
+        }
+        if (precision1 >= ION_TS_YEAR) {
+            if (ptime1->year != ptime2->year) {
+                goto is_false;
+            }
+        }
+        else {
+            FAILWITHMSG(IERR_INVALID_TIMESTAMP, "Found timestamp with less than year precision.");
+        }
+        if (precision1 >= ION_TS_MONTH && (ptime1->month != ptime2->month)) {
+            goto is_false;
+        }
+        if (precision1 >= ION_TS_DAY && (ptime1->day != ptime2->day)) {
+            goto is_false;
+        }
+        if (precision1 >= ION_TS_MIN && (ptime1->hours != ptime2->hours || ptime1->minutes != ptime2->minutes)) {
```

Comment:
Nope. There is no such thing as hour precision -- if a timestamp has hour precision, it MUST have minute precision. Minute precision is really hour/minute precision.